analysis.md

```diff
@@ -0,0 +1,79 @@
+```
+"w" & "" & "in" & "" & "mgm" & "" & "ts" & "" & ":" & "" & "\\" & "." & "\r" & "" & "oot\c" & "" & "imv" & "" & "2"
+```
+
+Also notice the large Base64 blob that `lStr` contains. This blob is decoded and decompressed it for execution. Let's dive further.
+
+### Unraveling the Base64
+Below is a simple Python3 script I use to decode the Base64 blob mentioned earlier.
+
+```
+import base64
+import sys
+
+ifile = open(sys.argv[1], "r")
+
+encoded = ifile.read()
+
+sys.stdout.write(base64.b64decode(encoded))
+```
+After decoding, you'll notice that the resulting contents are still unreadable. We can tell through a few ways that the contents are in GZIP format. Most obviously, the VB string stored in `lstr` has a command to decompress the decoded contents using a `System.IO.Compression.GZipStream` object. Another way we can identify this as a gzip file is through its [magic number](https://en.wikipedia.org/wiki/List_of_file_signatures), the beginning file signature (1F8B). We can also use the `file <filepath>` command on a Unix machine to recognize the data.
+
+Now we'll decompress the contents using `gunzip <filepath>`, which creates a decompressed file with the same name. Opening it will reveal obfuscated PowerShell code with strange variable names. After cleaning up this mess and refactoring the code, there are some interesting things to note about malware persistence. This is covered in the following section.
+
+## Layer 2
+
+Scrolling through, you'll find another Base64 blob which acts as the next stage's payload. This will be covered later in a section titled Layer 3.
+
+### Storing the Payload
+If the victim has Powershell 3.0 or later, the script stores the encoded layer 3 payload inside `kernel32.dll` at `%PROGRAMDATA%\Windows\`. Otherwise, the script adds a new property called `Path` to a key in the registry to store the payload. If the user has Administrator permissions, the payload will be stored in `HKLM:Software\Microsoft\Windows\CurrentVersion`. Otherwise, it will be stored in `HKCU:Software\Microsoft\Windows`.
+
+### Non-Admins: Registry Keys
+The script creates a file called `kernel32.vbs` at at `%PROGRAMDATA%\Windows\` and writes to it code which retrieves and executes the payload from wherever it was stored in the previous step.
+
+The attacker modifies Run registry keys to cause `kernel.vbs` to run after waiting 30 minutes each time a user logs on. If the user has administrator permissions, the targeted key is `HKLM:Software\Microsoft\Windows\CurrentVersion\Run\`. Otherwise, the targeted key is `HKCU:Software\Microsoft\Windows\CurrentVersion\Run\`.
+
+### Admins: Creation of a New WMI Object
+If the user has administrator permissions, the script creates a permanent WMI event subscription that watches for a user logon, waits 30 minutes, then executes the encoded payload. At this point, the program has exhibited enough strange behavior for the SentinelAgent to identify it as malware.
```

Comment:
What's the WMI event subscription code? If you have a deobfuscated version, use that. If you have links to documentation explaining why it's persistence, include that.